### ISSUE TITLE
Make Android / iOS version of AdGuard not to filter Telnet Bulletin Board System (BBS). (Feature request)

### ISSUE THREAD

**Issue by lancelot-moon** [AdguardTeam/AdguardForAndroid#1601]
I think it's better to open a new issue for the feature request.

ameshkov:
> Should we disable HTTPS filtering for these apps by default?

lancelot-moon:
> I disabled AdGuard's HTTPS filtering in advance.
And then I tried to login into PTT BBS via the default port setting 443 without SSH connection.
If using JPTT, I have to wait for 2 minutes for login.
The waiting time is too long.

> If using Mo PTT, I was unable to connect to PTT BBS server.
Mo PTT showed the error,
"Oh! The server has no response, please try again later."

> Disabling HTTPS filtering is absolutely not the solution.
A risk of being unable to connect to PTT BBS.

Привет ameshkov,

Here is the information about PTT BBS.

https://github.com/ptt/pttbbs
If you have advanced questions, you can try to ask them there.
https://github.com/ptt/pttbbs/issues

> mainly used in telnet / ssh protocol, currently web-interface is used only for reading articles, no other interactions.

> SSH connection: bbs@ptt.cc
Telnet unsafe connection: telnet://ptt.cc
Website: www.ptt.cc

AdGuard is unable to filter SSH connection, right?
How does AdGuard deal with Telnet protocol?
Telnet protocol:
https://ru.wikipedia.org/wiki/Telnet

The problem is unsafe Telnet connection with AdGuard.
I tried to disable HTTPS filtering.
But it's useless.
There is still the risk of being unable to connect to PTT BBS if using the default port 443 for mobile BBS apps.
You can try Mo PTT for login into PTT BBS as a guest with the default port 443 by yourself.
Please click "我沒有帳號，路過看看".
It means:
"I don't have an account. Visit as a guest."
<details><summary>Screenshot:</summary>

![image](https://i.imgur.com/vjUAbPU.png)
</details><br/>

I don't know if not filtering all Telnet BBS is a good thing.
However, at least, can you design that AdGuard doesn't filter the 3 servers of PTT Telnet BBS by default?
PTT Telnet BBS has the 3 associated servers.
telnet://ptt.cc <- (批踢踢實業坊) (Main server)
telnet://ptt2.cc <- (批踢踢兔) 
telnet://ptt3.cc <- (批踢踢參)

The allowed ports for connecting to PTT BBS are only 23, 443 and 3000~3010.
For example:
telnet://ptt.cc:443
telnet://ptt.cc:3005

> No, that's because AG does not filter connections to ports less than 80

Does it seem that AG also does not filter connections to ports above 3000?

The general default port setting of telnet is 23.

But most of mobile BBS apps use port 443 as the default port setting, such as JPTT, Mo PTT...etc.

A few BBS apps don't use unsafe Telnet connection.
They use SSH connection as the only way of connecting to PTT BBS, such as PTT+, PiTT, isPTT, PTTopia...etc.

Спасибо!

**Comment by ameshkov** [comment]
@lancelot-moon we'll solve it in v2.11, sorry for the late reply

**Comment by lancelot-moon** [comment]
Taiwanese net friends thank you very much!!!
The allowed ports for connecting to PTT BBS are only 23, 443 and 3000~3010.
If not filtering Telnet protocol or by default simply not filtering telnet://ptt.cc, telnet://ptt2.cc and telnet://ptt3.cc, it's a great news!

**Comment by lancelot-moon** [comment]
Hi,
Some Taiwanese people are trying new safe connection to PTT BBS.
https://github.com/iamchucky/PttChrome

An experimental safe connection to PTT BBS, compared with SSH.
It's WebSocket Protocol.
We won't need a BBS app.
Only use a browser for login into PTT BBS.
https://term.ptt.cc

I worry if AdGuard also filters WebSocket Protocol and causes the connection issue.
Just a share. XD

**Comment by Revertron** [comment]
Hm, I've installed Mo PTT, and if I click on "我沒有帳號，路過看看" I'm able to see many topics and posts. I don't speak Japanese, but I think it works perfectly from here (in root-proxy mode).

**Comment by lancelot-moon** [comment]
Hi,

It's Traditional Chinese, not Japanese.

I rarely use local proxy mode because my phone has no root.

Maybe you can use local VPN mode for the test.

If there is an PTT BBS account for login, users are unable to login into PTT BBS by default port 443 with AdGuard.
https://i.imgur.com/XqJvy7r.png

Need to modify port number to 23 or 3000~3010 in mobile PTT BBS app because AdGuard doesn't filter those port numbers.
Or simply use SSH connection.

If there is no account and then enter as a guest, I don't know why users can enter PTT BBS by default port 443 with AdGuard.

Maybe you need a PTT BBS account and then you'll understand my meaning.

Do you have an email or skype?
I maybe need to register a new account for AdGuard team's test.

**Comment by Revertron** [comment]
Yes, you can send me a letter to revertron (at) adguard.com with some credentials and I will check it.

**Comment by lancelot-moon** [comment]
Hi,

I have sent the id & pw to you.
Please check.

There are many mobile PTT BBS apps.
https://github.com/AdguardTeam/AdguardForAndroid/issues/1520#issue-264074297

Mo PTT & JPTT provides 2 connection methods.
1. Unsafe BBS Telnet connection
2. Safe SSH connection

I suggest that you install the 2 apps.
And you use **local VPN mode for the test.**
Please don't use local proxy mode.

JPTT:
https://play.google.com/store/apps/details?id=com.joshua.jptt

Mo PTT:
https://play.google.com/store/apps/details?id=mong.moptt&hl=zh_TW

If having a PTT BBS account...
With default port 443 and AdGuard VPN, JPTT needs to wait for around 2 minutes for login.
With default port 443 and AdGuard VPN, Mo PTT will finally show connection error after some time.

Some other mobile PTT BBS apps set SSH connection as the only way to PTT, such like PiTT app, PTT+ app and so on.
We don't use these kind of mobile PTT BBS for our test because AdGuard is unable to filter SSH connection.

We need to solve the issue between Unsafe BBS Telnet connection and AdGuard VPN.

Those PTT apps are Traditional Chinese interface.
Please feel free to contact with me if needing my help.

B/R

**Comment by lancelot-moon** [comment]
@Revertron 

http://pcman.ptt.cc/downloads.html

You can first install and run PCMan in Windows, which maybe help you to be familiar with PTT BBS.

PCMan:
http://pcman.ptt.cc/downloads.html
Interface:
<details><summary>Screenshot:</summary>

![image](https://i.imgur.com/uRAG4Wu.png)
</details><br/>

Address: ptt.cc
Don't input guest since that you have an account.
Input the id & pw which I lend you for login into PTT.

PTT BBS operation interface:
<details><summary>Screenshot:</summary>

![image](https://i.imgur.com/OBtj45f.png)
</details><br/>

**Comment by ameshkov** [comment]
@Revertron as I understand, when the port is set to 443, AG tries to filter it as an HTTPS connection. The problem is that inside of the tunnel the protocol is not HTTP, and AG simply breaks it.

We need to figure out how to solve that. The easiest solution would be to add affected domains to the HTTPS exclusions list.

**Comment by lancelot-moon** [comment]
Adding ptt.cc in the HTTPS exclusions list is useless.
Disabling HTTPS filtering is also useless.

@ameshkov 
I had lent a PTT BBS ID & Password to Revertron.
You can also use that account together.
Multi-login is available.
When showing the message, input "n" and then click "enter" on keyboard.
<details><summary>Screenshot:</summary>

![image](https://i.imgur.com/iG5T2XS.png)
</details><br/>
I suggest that you use PCMAN for login into PTT BBS on your Windows PC computer.
It's helpful to let you know what BBS is.

With AdGuard **local VPN mode**, I don't know why there are different results when login into PTT BBS as a member or a guest.

On iOS, mobile PTT BBS apps have no connection issue with AdGuard Pro via port 443.
On Android, mobile PTT BBS apps have the connection issue with AdGuard via port 443 when using unsafe BBS Telnet protocol for connecting.

**Comment by Revertron** [comment]
I have a great news for you, @lancelot-moon! We've nailed this problem. It seems that this BBS software is waiting from the server the first request instead of sending it's request to the server first as do all the "normal" apps and browsers.
The fix will be in next Nightly, very soon :)

**Comment by lancelot-moon** [comment]
Wow, so cool!!!
The issue tormented AdGuard's Taiwanese users for a long time.
You're superman! m(_  _)m
(*￣▽￣)/‧☆*"`'*-.,_,.-*'`"*-.,_☆ 

I will try the next Nightly version.
Thank you very much!

**Comment by lancelot-moon** [comment]
Hi,
I saw that AdGuard beta v2.11.31 is released.
Have you solve the BBS issue in AdGuard beta v2.11.31?
Or do I need to wait next AdGuard Nightly version?
Which is newer about AdGuard v2.11.32η and AdGuard beta v2.11.31?
Thanks.

**Comment by Revertron** [comment]
As I've said earlier, the fix will be in the next Nightly, sorry :-\

**Comment by lancelot-moon** [comment]
Sir,
Thank you very much!
I tried v2.11.43η .
Now, mobile PTT BBS apps can normally login via port 443 with AdGuard local VPN.

But their speed of login are delayed.
Without AdGuard local VPN:
JPTT takes 1 second for login.
MoPTT takes 2 seconds for login.

With AdGuard local VPN:
They need about 6 seconds for login.

Is it possible to improve the time of login into PTT BBS with AdGuard local VPN?

Thanks a lot!
